src/UsblParser.hpp

```diff
@@ -3,114 +3,72 @@
-             * @return The extracted int
-             */
-            static int parseInt(uint8_t const* data, size_t const size, std::string const command);
-            /*
-             * The function interpretes the response as OK.
-             * If the respnse isn't an OK the function throws an Exception.
-             * @param[in] data The data in a uint8 array to analysing
-             * @param[in] size The length of data
-             */
-            static void parseOk(uint8_t const* data, size_t const size);
-            /*
-             * The function interpretes the string as a connection status and
-             * returns a enum with the status.
-             *
-             * @param[in] command The command to analysing
-             * @return The connection status in a enum
-             */
-            static ConnectionStatus parseConnectionStatus(std::string const command);
-            /*
-             * The function interpretes the string as a incoming instant message and returns
-             * the extracted instant message
-             *
-             * @param[in] command The command to analysing
-             * @return The instant message in a struct
-             */
-            static ReceiveInstantMessage parseIncomingIm(std::string const s);
-            /*
-             * The functions interpretes the string as a position and
-             * returns a struct with the position.
-             *
-             * @param[in] command The command to analysing
-             * @return The position in a struct
-             */
-            static Position parsePosition(std::string const command);
-            static Position parseRemotePosition(std::string const s);
-            /*
-             * The funcion extracts the string in the response and returns the string.
-             * You have to interprete the string anymore.
-             * The function throws an exception, if the response isn't fit to the command.
-             *
-             * @param[in] data The data in a uint8 array to analysing
-             * @param[in] size The length of data
-             * @param[in] command The command to fit the data with this command 
-             * @return The extracted string
-             */
-            static std::string parseString(uint8_t const* data, size_t const size, std::string const command);
-
-            static std::string parsePhyNumber(std::string const s);
-            static std::string parseMacNumber(std::string const s);
-    };
+class UsblParser
+{
+private:
+    std::vector<std::string> splitValidate(std::string const &buffer, const char* symbol, size_t const parts);
+    //            static std::vector<std::string> validateResponse(std::string const s);
+    //            static int getInt(std::string s);
+
+public:
+    UsblParser();
+    ~UsblParser();
+
+
+    // Given buffer, determine if it's a Notification and returns it's kind.
```

Comment:
These comments are not properly formatted for doxygen. Use

~~~
/** Summary line
 *
 * long comment
 */
~~~